1224-user_administration/1424-administrator-configures-automatic-creator-role.feature

```diff
@@ -0,0 +1,14 @@
+    @1424
+    Scenario Outline: Institution Administrator configures automatic Creator role
+        Given that the User is logged in as Institution Administrator
+        And they are on the User Administration Page
+        And they see a list of checkboxes in the Registrator section
+        And they see each checkbox has corresponding <Option> text
+        And they see that none of the checkboxes are selected
+        When they select <Option>
+        And they click the Save button
+        Then they see a Notification that the new values are persisted
+        Examples:
+            | Option                             | Category                  |
+            | Scientific employees               | member, employee, faculty |
+            | Techincal administrative employees | member, employee, staff   |
```

Comment:
Kan ikke ha Category som eksempel her når det ikke er brukt til noe. Enten må det fjernes fra eksempel, eller så må det bli brukt. Synes Orestis sitt forslag var bra jeg.

Reply:
enig i at det enten må bort eller endres, men jeg syns det er litt i overkant å gi tilbakemelding av typen `Users with Feide category member, employee, faculty have got the Creator role`. Jeg tror ikke det vil gi noen mening for vanlige Institution admins siden jeg tviler på at de vet hva Feide-kategori er for noe.
Sånn jeg ser det er det som ligger i <Category> irrelevant for brukeren, men det er de verdiene appen må håndtere, så jeg ser ikke helt hvordan det skal komme med i scenarioet...

Reply:
Men selv om det står at brukeren skal få tilbakemelding om at de har fått Creator-rollen, så betyr det ikke bokstavelig talt at det skal stå noe om `member, employee, faculty`. Det er vel mer for backend sin del at det kanskje burde være med...

Hva med dette da?
```
Then they see a Notification that the <Option> have got the Creator role
```
også ta bort Category? 

@axthosarouris?

Reply:
Jeg syns det gir mening ja (for frontend), men hvordan skal den som skal implementere dette på backend da vite at dette henger sammen med det som nå står i `<Category>`?
(sorry for at jeg er vrang, men sliter med å forstå hvordan et scenario egentlig skal se ut)

Reply:
ja, det er det som er så vanskelig. Men hittil har vi kun beskrevet frontend-logikk, også har backend "funnet ut" hva som egentlig skjer, også har de skrevet sine egne scenario på sitt repo for å forklare nøyere hva som skjer.

Reply:
* The first problem I see is that scenario title says that something about Creator role but the effect is that some values are being saved. The effect should be that the selected employee categories will be automatically assigned the Creator role.

*I think it correct that we should not have things in the scenario that are not in use. If they are not in use then the product owner things that it is specified and automatically checked when it is not. 
If an example value is not in a scenario it should be either be removed or included in the scenario. 

* Here the Options list plays the role of displaying all the Options (listing) and selecting one value.
I think we are using the Examples table as both a listing of options and as variables to be replaced and that is why we have trouble with this scenario. If you replace the <Options> with one of the example the sentence 
```
they see each checkbox has corresponding <Option> text
```
does not make sense. 
Example:
```
they see each checkbox has corresponding  Scientific employees  text
they see each checkbox has corresponding   Techincal administrative employees  text
```
This is important because when the scenario will fail we will see only the replaced text and not the whole scenario. Remember that the scenario is code. 

* It can be that the scenario should be split in what it is shown and what happens when a person makes a choice. Then it can be easier to describe both scenarios in a better way.

* What does it mean that "the new values are persisted"? Which values are going to be persisted? The actual effect is not that something will be persisted but that these employee categories will be assigned the role of Creator.


Consider this alternative :
```
And they see a list of checkboxes in the Registrator section with the labels:
   | Scientific employees     |
   | Techincal administrative employees |
....
When they select the checkbox with with text <Option> 
And they click the Save button
Then they see a Notification that the Categories <Category> will be assigned a Creator role